### ISSUE TITLE
How many epochs will be take when training celeba?

### ISSUE THREAD

**Issue by weitaoatvison** [openai/glow#14]
Hi, I'm trying to train a model on celeba datasets with your code. And I trained the model with the command line below:
mpiexec -n 8 python train.py --problem celeba --image_size 256 --n_level 6 --depth 32 --flow_permutation 2 --flow_coupl    ing 0 --seed 0 --learntop --lr 0.001 --n_bits_x 5

I just modify the gpu numbers to 8 because I only have a 8 cards server. Now, I have trained for 4days with 1300epochs,  however, the results seems not good when I check the example images generate by code. So, could you tell me how many epochs it takes when you trained on celeba dataset? @prafullasd

**Comment by voa18105** [comment]
Same for me, it does not converge to results presented in the article. How long it should be trained?

**Comment by pclucas14** [comment]
Running into the same issue with mnist :/

**Comment by pclucas14** [comment]
@weitaoatvison @voa18105 could you post pictures of the samples you get ?

**Comment by voa18105** [comment]
Sure, here my beauties :-) ![screenshot_20180713-134952](https://user-images.githubusercontent.com/32035001/42928217-5ea40ebc-8b3f-11e8-9d23-512254b999d8.jpg)

**Comment by weitaoatvison** [comment]
@voa18105 My result is similar to yours. 1300epoch  
![epoch_1300_sample_5](https://user-images.githubusercontent.com/18735019/42934690-4f1cdccc-8b7a-11e8-8713-133390202ae8.png)

**Comment by voa18105** [comment]
Actually, your's is better. So, it improves.

**Comment by dpkingma** [comment]
@voa18105 @weitaoatvison 
The samples we show in the paper are after about 4000 training epochs. As shown in figure 8 of the paper the (subjectively) most realistic-looking samples were obtained with lower temperature. The samples from figure 1 are with a temperature of 0.7. The training script creates samples with various temperatures (number 1-, so make sure to look at the right ones, namely "*_sample_4.png". See [https://github.com/openai/glow/blob/master/train.py#L47](https://github.com/openai/glow/blob/master/train.py#L47.)

**Comment by dpkingma** [comment]
@voa18105 @weitaoatvison 

P.S. I encourage you to look at temperature=0.7 samples at 1300 epochs. They should look better than the one shown above, which was probably with temperature=1.0.

Cheers

**Comment by winwinJJiang** [comment]
Hi, anyone reproduced the results and would like to post it?

Thanks,

2018-07-19 5:37 GMT-04:00 weitaoatvison <notifications@github.com>:
